### ISSUE TITLE
Q: How to replace createSearchParams

### ISSUE THREAD

**Issue by tordans** [nanostores/router#30]
In https://github.com/nanostores/router/blob/main/CHANGELOG.md?plain=1#L10 I read that 
> * Removed `createSearchParams` in favor of `route.search` of normal router.

I have this Astro App which uses `export const $searchParams = createSearchParams()` to create a store that is stored in the URL. https://github.com/osmberlin/www.osm-verkehrswende.org/blob/main/src/components/BaseMap/store.ts#L23

How would I migrate that to route.search? I am not using the router in Astro, I am looking for something like an `atom`, but in the URL.

**Comment by ai** [comment]
There are two options:

1. Copy `createSearchParams()` implementation to your project. It is very simple store, you don’t to need some official library for that. Or you can create a support this library.
2. Use normal `createRouter()` (you can use single route if you don’t need `pathname` routing) and read `value.search` of the router store.

**Comment by tordans** [comment]
Thanks for the hints @ai.

I tried suggestion (2) today but that is not ideal for this use case because it requires to specify the path of each route which uses my map component because the component needs to use `openPage($router, 'NAME', undefined, { ...newParams })`. I really just want to use the search params just like any other store without thinking about the URL and path.

So I went with suggestion (1) and copied the version 0.13.0 code in https://github.com/osmberlin/www.osm-verkehrswende.org/commit/3d2e27b4fa7f427e3f72af41561d025857ec57c5 (with some minor TS adjustments).
It looks like this works fine.

I with there was something like https://github.com/47ng/nuqs for Astro/Plain React which work great in this use case where one wants to use a global state but store it in the URL.
